modules/01-kafka-on-openshift.md

```diff
@@ -98,4 +98,75 @@ The `KafkaUser` resources are described in the `kafka/kafka-users.yaml` file.
+- Deployment - specification of the Prometheus Operator deployment
+- ServiceAccount
+- Service
+
+This file is generated so you should modify it. You should remove all the `namespace: default` values except of one in `ClusterRoleBinding`. Here you should put your namespace.
+When deploying on OpenShift you should also remove `spec.securityContext` from the `Deployment`.
+
+After these modifications you can use
+```shell
+oc apply -f bundle.yaml
+```
+
+### Deploying Prometheus and Alertmanager
+When the Prometheus Operator is deployed, you can send a `Prometheus` and `Alertmanager` resource to the OpenShift server API and Prometheus Operator should deploy desired number of Prometheus and Alertmanager servers.
+
+Before we do that, we need to prepare a `Secret`s resources which are used by servers. These `Secret`s contain additional configuration for Prometheus and Alertmanager servers.
+
+```shell
+TBD: adding entire prometheus dir to this repo?
+oc create secret generic additional-scrape-configs --from-file=additional-properties/prometheus-additional.yaml
+oc create secret generic alertmanager-alertmanager --from-file=alertmanager.yaml=alertmanager-config/alert-manager-config.yaml
+```
+
+Now we can apply all files from the `install` folder.
+```shell
+oc apply -f install
+```
+
+By this, these files are applied:
+- alert-manager.yaml - specificaion of the `Alertmanager` resource and the `Service` which is pulling alerts from the Prometheus server
+- prometheus.yaml - specification of the `Prometheus` resource including a RBAC resources
+- prometheus-rules.yaml - specification of the `PrometheusRule`. These are conditions which when violated, an alert is fired
+- strimzi-service-monitor.yaml - specification of the `ServiceMonitor` resource where are described jobs for scrapping a metrics
+
+Now the Prometheus and Alertmanager pods should be created.
+
+### Deploying Grafana
+Now we will deploy the last segment of the chain. Grafana is deployed by applying `Deployment` resource. 
+
+```shell
+TBD: add grafana folder?
+oc apply -f grafana.yaml
+```
+
+## Setting up the Grafana server
+You can set Graphana in UI or by using its API. We will for use an API way for the sake of this workshop simplicity.
+As first step we need to create a datasource. 
+
+```shell
+curl --user admin:admin 'http://'$(oc get svc grafana -o=jsonpath='{.spec.clusterIP}')':3000/api/datasources' -X POST -H 'Content-Type: application/json;charset=UTF-8' --data-binary '{"name":"Prometheus","isDefault":true ,"type":"prometheus","url":"http://'`oc get pod prometheus-prometheus-0 --template={{.status.podIP}}`':9090","access":"proxy","basicAuth":false}'
+```
+
+Then we can create a dashboard which is receiving data from this datasource.
+
+```shell
+TBD: copy strimzi-kafka.json
+curl --user admin:admin -X POST 'http://'$(oc get svc grafana -o=jsonpath='{.spec.clusterIP}')':3000/api/dashboards/import' -d @strimzi-kafka.json --header "Content-Type: application/json"
```

Comment:
as I mentioned offline the dashboards are slightly different to be used via API. You should put them here in the repo.